File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int total = 0;

        for (int num : nums) {
            total += num;
        }

        if (total < x) return -1;
        if (total == x) return nums.length;

        int left = 0;
        int sum = 0;
        int count = -1;
        int target = total - x;

        for (int right = 0; right < nums.length; right++) {
            sum += nums[right];

            while (sum > target && left <= right) {
                sum -= nums[left];
                left++;
            }

            if (sum == target) {
                count = Math.max(count, right - left + 1);
            }
        }

        return count == -1 ? -1 : nums.length - count;
    }
}

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int n=nums.size();
        long long total=accumulate(nums.begin(),nums.end(),0LL);
        if(total<x) return -1;
        if(total==x) return n;
        int i=0, j=0, sum=0, maxi=0;
        int target=total-x;
        while(j<n){
            sum+=nums[j];
            while(j>=i && sum>target){
                sum-=nums[i];
                i++;
            }
            if(sum==target) maxi=max(maxi,j-i+1);
            j++;
        }
        return maxi==0 ? -1 : n-maxi;
    }
};